### ISSUE TITLE
MEMO OF THINGS TO DO

### ISSUE THREAD

**Issue by dalessioluca** [spacetx/spacetx-research#27]
0. produce one large segmentation of smFISH_OLEH and VISIUM
1. check batch_norm in UNET (currently is not there)
2. reflection padding in UNET (currently is not there) or nothing and then prediction on a smaller region?
3. the graph is not a K_NN graph. is that ok? Optimize the radius. It seems that larger is better (i.e. 5 is better than 2). To evaluate this systematically you need to make plots of N_OBJECTS vs RESOLUTION parameters. Hopefully for large radius we will see a plateau
4.  is greedy modularity optimization the thing we are interested in? TIM suggests: If you aren’t committed to greedy modularity maximization, one of the fastest libraries that will get you community detection (using Stochastic Block Models) is graph tool (https://graph-tool.skewed.de/). It’s c++ underneath (using Boost I believe), so it is very fast. The tradeoff is that it can be a huge pain in the ass to install, though I have heard it has recently been simplified.
5. the graph is partitioned in disconnected components. Is there an advantage in treating each connected component separately. Is community detection faster? Can I use the same resolution parameters for all the different disconnected components
6. loss function optimization. It seems that the best loss function was the one in 
folder: /home/jupyter/REPOS/spacetx-research/NEW_ARCHIVE/merfish_june22_v2
commit 39d6bf2c968381b155a7d8d868b9f4de1f19184f
Change master implementation back to that one. Try to understand the differences.

**Comment by dalessioluca** [comment]
I might have a bug in graph construction since the plot of sum_j e_ij does not look right.
I want a density clustering algorithm on a graph. Read: http://dm.uestc.edu.cn/wp-content/uploads/paper/Graph%20Clustering%20with%20Local%20Density-Cut.pdf

**Comment by dalessioluca** [comment]
Do a watershed based on d_i = sum_j e_ij

**Comment by dalessioluca** [comment]
measure distance based || pi_i - pi_j ||^2 not e_ij = < pi_i | pi_j >
